src/twiliosim.cr

```diff
@@ -31,85 +36,155 @@ class Twiliosim::Server
       response.to_json(context.response)
     when %r(/Accounts/(.+)/Calls.*)
       account_sid = $1
-      body = request.body
-      unless body
-        plain_response(context, 400, "Request body is missing")
-        return
-      end
-      body = body.gets_to_end
-      unless body
-        plain_response(context, 400, "Request body is missing")
-        return
-      end
-      if body.blank?
-        plain_response(context, 400, "Request body is missing")
-        return
-      end
-      params = HTTP::Params.parse(body)
-      unless params.has_key?("Url")
-        plain_response(context, 400, "Body param 'Url' is missing")
-        return
-      end
-      verboice_url = params["Url"]
-      unless params.has_key?("From")
-        plain_response(context, 400, "Body param 'From' is missing")
+      handle_call_request(context, account_sid)
+    else
+      plain_response(context, 404, "404 Not Found")
+    end
+  end
+
+  private def handle_incoming_phone_numbers_request(context)
+    context.response.status_code = 200
+    context.response.content_type = "application/json"
+    response = {"sid" => UUID.random().to_s()}
+    response.to_json(context.response)
+  end
+
+  private def handle_call_request(context, account_sid)
+    body_params = get_body_params(context.request.body)
```

Comment:
I think it would be best if get_body_params raises exception on invalid arguments. That will generate a 500 directly. Or you can rescue the exception and render a 400 if needed.

The `get_body_params` han be named `call_params` and return a named tuple with only to/from/verboice_url.